### ISSUE TITLE
[now-cli] Now cli 16.3.0 `--prod` option does not alias domain when build succeeded

### ISSUE THREAD

**Issue by glemmaPaul** [vercel/vercel#3107]
Hi everyone,

First of all, amazing service and this is really the first time I've found an issue in a process that was overall very seamless, thank you!

Now I did get an issue with updating `now-cli` to 16.3.0, and to my surprise my build does not get to be aliased to my custom domain. Going back to `now-cli@16.2.0` fixes my issue completely.

My command is the following:
```
now --prod --local-config ./.now/now.dev.json
```
My config is:
```
{
  "name": "website-dev",
  "alias": "domain.com",
  "version": 2,
  "build": {
    "env": {
      // hidden
    }
  },
  "builds": [{
    "src": "next.config.js",
    "use": "@now/next"
  }]
}
```

Is there something I need to update to support the newer now-cli updates?

Thanks in advance, let me know if you need more info from me!

**Comment by leo** [comment]
Thank you for the compliment and your report!

Considering the behaviour changed between the last version and the current version, I will investigate this problem and ensure that we return to the previously working state.

I'll keep you updated on this thread.

**Comment by leo** [comment]
We have found the error and are working on a fix. We're also adding a test for this specific case, so that it won't ever happen again.

As in turns out, it only happens if `--local-config` is used.

**Comment by glemmaPaul** [comment]
Thank you @leo for the quick response on this!

**Comment by styfle** [comment]
Thank you for the bug report!

We released a fix for `--local-config` on the canary channel.

Can you try running `npm i -g now@canary` to confirm this version works for you? Thanks!

**Comment by samuelroy** [comment]
@styfle I got the same issue with the latest canary version `16.1.4-canary.13`

It works on the second run when there's no files to sync (so no build step):

![image](https://user-images.githubusercontent.com/10351822/66071448-75a05980-e553-11e9-91c0-983d69f0c969.png)

**Comment by leo** [comment]
Thank you for letting us know @samuelroy! We'll take a look.

**Comment by theBliz** [comment]
@samuelroy how did you change the `now.production.json` file to run with the latest canary version `16.1.4-canary.13`?

My now.production.json looks like this
```
{
    "version": 2,
    "name": "my-production-name",
    "alias": "myproductiondomain.com",
    "builds": [{ "src": "package.json", "use": "@now/next" }]
}
```

**Comment by samuelroy** [comment]
mine looks like this:

```json
{
  "name": "example",
  "version": 2,
  "regions": ["bru"],
  "scope": "team",
  "alias": "example.domain.com",
  "env": {
    "URL_API": "https://api-dot-example.appspot.com",
    "OTHER": "test",
  },
  "build": {
    "env": {
      "NODE_ENV": "production"
    }
  }
}
```

**Comment by theBliz** [comment]
@samuelroy thanks for sharing! I've solved #3107 by using `now@canary`

```
npm install --global --unsafe-perm now@canary
now --prod --token $ZEIT_TOKEN --local-config .now/now.production.json
```
and `now.production.json`
```
{
    "version": 2,
    "name": "my-production-name",
    "alias": "myproductiondomain.com",
    "builds": [{ "src": "package.json", "use": "@now/next" }]
}
```
@styfle when should we expect this fix to be merged on the main release?
also, where can I check the main differences between main and canary?

**Comment by styfle** [comment]
The stable channel has been updated with this fix, thanks

https://github.com/zeit/now/releases/tag/now%4016.3.1

Stable is master branch https://github.com/zeit/now/commits/master
Canary is canary branch https://github.com/zeit/now/commits/canary
